asmcomp/comballoc.ml

```diff
@@ -34,25 +34,23 @@ let rec combine i allocstate =
       (i, allocated_size allocstate)
   | Iop(Ialloc { words = sz; _ }) ->
       begin match allocstate with
-        No_alloc ->
-          let (newnext, newsz) =
-            combine i.next (Pending_alloc(i.res.(0), sz)) in
-          (instr_cons_debug (Iop(Ialloc {words = newsz; spacetime_index = 0;
-              label_after_call_gc = None; }))
-            i.arg i.res i.dbg newnext, 0)
-      | Pending_alloc(reg, ofs) ->
-          if ofs + sz < Config.max_young_wosize * Arch.size_addr then begin
-            let (newnext, newsz) =
-              combine i.next (Pending_alloc(reg, ofs + sz)) in
-            (instr_cons (Iop(Iintop_imm(Iadd, ofs))) [| reg |] i.res newnext,
+      | Pending_alloc(reg, totalsz)
+          when totalsz + sz < Config.max_young_wosize * Arch.size_addr ->
```

Comment:
Q: Why is the multiplication by `Arch.size_addr` here?  (Yes, I know it was there before...)

A: It looks like I messed up when adding the inline record field called `words`.  It should be called `bytes` instead.  I will make a pull request to fix this.  (This shows the value of inline record fields rather than anonymous constructor arguments, especially when units of measure are concerned that do not have their own types.)